src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,87 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int size = 0;
+    private int threshold;
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
+
+    private int hash(K key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode()) % table.length;
+    }
 
     @Override
     public void put(K key, V value) {
+        int index = hash(key);
 
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if ((key == null && node.key == null) || (key != null && key.equals(node.key))) {
+                node.value = value;
+                return;
+            }
+        }
+
+        Node<K, V> newNode = new Node<>(key, value, table[index]);
+        table[index] = newNode;
+        size++;
+
+        if (size >= threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = hash(key);
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if ((key == null && node.key == null) || (key != null && key.equals(node.key))) {
+                return node.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        int newCapacity = table.length * 2;
+        Node<K, V>[] newTable = new Node[newCapacity];
```

Comment:
The diamond operator is used without type parameters here as well. It should be parameterized with types for clarity and type safety, like `new Node<K, V>[newCapacity]`. However, due to type erasure, you'll need to create an array of raw type and cast it: `(Node<K, V>[]) new Node<?>[newCapacity]`.